File: go/raindrops/raindrops.go
package raindrops

import "strconv"

// Convert an integer to raindrops string
func Convert(i int) string {
	r := ""
	if i%3 == 0 {
		r += "Pling"
	}
	if i%5 == 0 {
		r += "Plang"
	}
	if i%7 == 0 {
		r += "Plong"
	}
	if r == "" {
		r = strconv.Itoa(i)
	}
	return r
}

File: ruby/raindrops/raindrops.rb
# Raindrops : slighty better at doing the sounds
class Raindrops
  PRIME_TO_SOUND = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong'
  }.freeze

  def self.convert(int)
    sounds = PRIME_TO_SOUND
             .select { |prime, _| (int % prime).zero? }
             .values

    sounds.any? ? sounds.join : int.to_s
  end
end
